### DISCUSSION TITLE
How to get a specify ResourceDictionary

### DISCUSSION THREAD

**Discussion by Yopler** [HandyOrg/HandyControl#1009] [Q&A] [answered]
Hello 🖐,

In a goal to share some of my modification of the style of HandyControl (that I really like btw 😊). In order to not have a file of 2000 lines, I separate my modification in differents files and then have a Theme.xaml where I merge all my ResourceDictionary. To achieved my modification I need to access to specify RessourceDictionnary of Handy and not get each time the 13000 line Handy's Theme.xaml 😅
Here an exemple of what I tried (that not working but help for context) : 
My **Theme.xaml** : 
```xaml
<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
    <ResourceDictionary.MergedDictionaries>
        <ResourceDictionary Source="pack://application:,,,/HandyControl;component/Themes/Theme.xaml"/>
        <ResourceDictionary Source="pack://application:,,,/HandyControl;component/Themes/SkinDark.xaml"/>
        <ResourceDictionary Source="Color.xaml"/>
        <ResourceDictionary Source="Styles/Navigation.xaml"/>
    </ResourceDictionary.MergedDictionaries>
</ResourceDictionary>
```

My custom modification **Navigation.xaml** :
```xaml
<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
                    xmlns:theme="https://handyorg.github.io/handycontrol">
    <ResourceDictionary.MergedDictionaries>
        <ResourceDictionary Source="pack://application:,,,/HandyControl;component/Themes/Styles/Base/ListBoxBaseStyle.xaml"/>
    </ResourceDictionary.MergedDictionaries>

    <Style x:Key="NavigationListBox" TargetType="ListBox" BasedOn="{StaticResource ListBoxBaseStyle}">
        <Setter Property="BorderThickness" Value="0"/>
        <Setter Property="Background" Value="Transparent"/>
    </Style>

    <Style x:Key="NavigationListBoxItem" TargetType="ListBoxItem" BasedOn="{StaticResource ListBoxItemBaseStyle}">
        <Setter Property="Background" Value="Transparent"/>
        <Setter Property="BorderThickness" Value="0,0,0,2"/>
        <Setter Property="Margin" Value="5,0"/>
        <Style.Triggers>
            <Trigger Property="IsMouseOver" Value="True">
                <Setter Property="Opacity" Value="0.7"/>
                <Setter Property="Background" Value="Transparent"/>
            </Trigger>
            <Trigger Property="IsSelected" Value="True">
                <Setter Property="Background" Value="Transparent"/>
                <Setter Property="BorderBrush" Value="{StaticResource PrimaryBrush}"/>
            </Trigger>
        </Style.Triggers>
    </Style></ResourceDictionary>
```

When I use this solution to use the `ListBoxBaseStyle`, I got an error : 

> Can't find the ressource 'themes/styles/base/listboxbasestyle.xaml'

![image](https://user-images.githubusercontent.com/45144437/136570070-4cc0d192-e676-4cd6-b2fd-2ab3f81ef1f8.png)

I have also try the Handy's `SharedResourceDictionnary
`
Do you have any idea how to access the `ListBoxBaseStyle `ResourceDictionnary ?

**Comment by Yopler** [comment, marked as answer]
@ghost1372 , @NaBian ?

**Reply by ghost1372** [reply]
So you want to use only the styles you need? This will not work Because styles depend on each other
for example in listview style we need, brushes, colors, converters, effects, geometries and... that Each of these resources is located separately
https://github.com/HandyOrg/HandyControl/tree/master/src/Shared/HandyControl_Shared/Themes/Basic

**Reply by Yopler** [reply]
Thanks for your response. 

I understand, but maybe I can import all the style needed ! I see that HandyControl used to do that (see commit : 39b1296727410a4cc48da51b0571f0ce5482e411). That's why I was asking ! ^^
And my main problem is that I can't access the ResourceDictionnary (like it doesn't exist)

**Reply by ghost1372** [reply]
As I said, what you want will not work, 
Commit you mentioned is no longer used and It was written before we used the xaml combine.
I Removed MergedDictionaries, see [here](https://github.com/HandyOrg/HandyControl/commit/d2642566f0c6123fa74f19b35803a2403bd8287c)
To increase performance, we merge all dictionaries into a theme file

**Reply by Yopler** [reply]
Ok thanks for answering ! 
I has been seen that you remove them but I was really hoping that's will still working.
Thanks 😄
